packages/file_selector/file_selector_android/android/src/main/java/dev/flutter/packages/file_selector_android/FileUtils.java

```diff
@@ -0,0 +1,198 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/*
+ * Copyright (C) 2007-2008 OpenIntents.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file was modified by the Flutter authors from the following original file:
+ * https://raw.githubusercontent.com/iPaulPro/aFileChooser/master/aFileChooser/src/com/ipaulpro/afilechooser/utils/FileUtils.java
+ */
+
+package dev.flutter.packages.file_selector_android;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Environment;
+import android.provider.DocumentsContract;
+import android.provider.MediaStore;
+import android.webkit.MimeTypeMap;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.UUID;
+
+public class FileUtils {
+
+  /** URI authority that represents access to external storage providers. */
+  public static final String EXTERNAL_DOCUMENT_AUTHORITY = "com.android.externalstorage.documents";
+
+  /**
+   * Retrieves path of directory represented by the specified {@code Uri}.
+   *
+   * <p>Intended to handle any cases needed to return paths from URIS retrieved from open
+   * documents/directories by starting one of {@code Intent.ACTION_OPEN_FILE}, {@code
+   * Intent.ACTION_OPEN_FILES}, or {@code Intent.ACTION_OPEN_DOCUMENT_TREE}.
+   *
+   * <p>Will return the path for on-device directories, but does not handle external storage
+   * volumes.
+   */
+  @NonNull
+  public static String getPathFromUri(@NonNull Context context, @NonNull Uri uri) {
+    String uriAuthority = uri.getAuthority();
+
+    if (uriAuthority.equals(EXTERNAL_DOCUMENT_AUTHORITY)) {
```

Comment:
`uriAuthority` can be null, I'd recommend swapping the order here to handle that case automatically
so
```suggestion
    if (EXTERNAL_DOCUMENT_AUTHORITY.equals(uriAuthority)) {
```